src/Emulator/HLE/Libraries/LibKernel/FileSystem/file_system.cpp

```diff
@@ -1,13 +1,51 @@
 #include "file_system.h"
 #include <debug.h>
 #include <Util/log.h>
+#include <Emulator/Util/singleton.h>
+#include "Emulator/Host/fs.h"
+#include <Core/PS4/HLE/ErrorCodes.h>
+
 
 namespace Emulator::HLE::Libraries::LibKernel::FileSystem {
 constexpr bool log_file_fs = true;  // disable it to disable logging
 
-int PS4_SYSV_ABI sceKernelOpen(const char* path, int flags, u16 mode) { 
+int PS4_SYSV_ABI sceKernelOpen(const char* path, int flags, u16 mode) {
 	LOG_INFO_IF(log_file_fs, "sceKernelOpen path = {} flags = {} mode = {}\n", path, log_hex_full(flags), log_hex_full(mode));
-	return 0; 
+
+	bool isDirectory = (flags & SCE_KERNEL_O_DIRECTORY) != 0;
+    bool createFileOrDir = (flags & SCE_KERNEL_O_CREAT) != 0;
+
+	auto* h = singleton<Emulator::Host::Fs::HandleTable>::instance();
+    auto* mnt = singleton<Emulator::Host::Fs::MntPoints>::instance();
+
+    u32 handle = h->createHandle().value();//TODO check if overflows
```

Comment:
Το handle εδώ γίνεται leak άμα το open call κανει fail. Πρεπει ειτε να το κλεινεις on failure ειτε να μην το δημιουργεις αν δεν ειναι σιγουρο οτι πετυχε

Reply:
if (h->createHandlle().has_value()) {
    handle = h->createHandle().value();
} else {
    return SCE_KERNEL_ERROR_EMFILE; //file descriptors is full error(?)
}

done?

Reply:
Αυτο ειναι το ενα μερος του.
Αλλα πες οτι πχ ενα file εν τελει δεν υπαρχει και το O_CREAT δεν ειναι set

Ο τωρινος κωδικας θα δημιουργησει ενα handle ετσι και αλλιως και ας αποτυχει το `open`, και ετσι κανει consume slots ενω δεν πρεπει. Για αυτο πρεπει ειτε οταν κανε fail το handle να γινεται free, η απλα να μην γινεται αλλοκατε